Mathlib/Algebra/Ring/Semireal/Defs.lean

```diff
@@ -28,19 +27,16 @@ semireal.
 variable (R : Type*)
 
 /--
-A semireal ring is a non-trivial commutative ring (with unit) in which `-1` is *not* a sum of
-squares. Note that `-1` does not make sense in a semiring. Below we define the class `IsSemireal R`
-for all additive monoid `R` equipped with a multiplication, a multiplicative unit and a negation.
+A semireal ring is a commutative ring (with unit) in which `-1` is *not* a sum of
+squares. We define the class `IsSemireal R` for all structures equipped with
+a multiplication, an addition, a multiplicative unit and an additive unit.
 -/
 @[mk_iff]
-class IsSemireal [AddMonoid R] [Mul R] [One R] [Neg R] : Prop where
-  non_trivial         : (0 : R) ≠ 1
-  not_isSumSq_neg_one : ¬IsSumSq (-1 : R)
-
-@[deprecated (since := "2024-08-09")] alias isSemireal := IsSemireal
-@[deprecated (since := "2024-08-09")] alias isSemireal.neg_one_not_SumSq :=
-  IsSemireal.not_isSumSq_neg_one
-
-instance [LinearOrderedRing R] : IsSemireal R where
-  non_trivial := zero_ne_one
-  not_isSumSq_neg_one := fun h ↦ (not_le (α := R)).2 neg_one_lt_zero h.nonneg
+class IsSemireal [Add R] [Mul R] [One R] [Zero R] : Prop where
+  not_IsSumSq_neg_one (a : R) (ssa : IsSumSq a) : ¬ (1 + a = 0)
```

Comment:
```suggestion
  not_isSumSq_neg_one (a : R) (ssa : IsSumSq a) : 1 + a \ne 0
```
or maybe
```suggestion
  add_one_ne_zero_of_isSumSq (a : R) (ssa : IsSumSq a) : a + 1 \ne 0
```

Reply:
Looks like you forgot to apply this?

Reply:
I just checked and I'm pretty sure it's correct as of now.

Reply:
Then maybe you forgot to push. I still see `¬ (1 + a = 0)` instead of `1 + a ≠ 0`

Reply:
Oh sorry, I only saw the name change, my bad

Reply:
Fixed